internal/server/recon/recognize.go

```diff
@@ -170,60 +174,67 @@ func getItemsForSpans(spans []string, query string, span2item map[string]*pb.Rec
 }
 
 // Splits a query by a span into a list of parts like: non-span part, span part,
-// non-span part, span part, etc. If the query does not contain the span,
-// returns a list with the query as the only item.
+// non-span part, span part, etc. and should only split on complete words. If
+// the query does not contain the span, returns a list with the query as the
+// only item.
 // Examples
 // query: "ab cd ef ab g", span: "ab", result: ["ab", "cd ef", "ab", "g"]
 // query: "ab cd ef g", span: "jk", result: ["ab cd ef g"]
 func splitQueryBySpan(query string, span string) []string {
-	splits := strings.Split(query, span)
-	// if span is not found in query, return just the query
-	if len(splits) < 2 {
-		return []string{query}
-	}
-	queryParts := []string{}
-	// prefix to add to the next split to be processed
-	nextPartPrefix := ""
-	// go through all the splits and add the span between each split if it was a
-	// valid split (we only want to split on complete words)
-	for i, split := range splits {
-		queryPart := nextPartPrefix + split
-		// if this is the last item in the list, add the query part and finish
-		if i == len(splits)-1 {
-			queryParts = append(queryParts, strings.TrimSpace(queryPart))
-			break
-		}
-		// Whether or not to skip adding anything in this round. We want to skip if
-		// this was not a valid split.
-		shouldSkipAdding := false
-		// if the current split is not empty and does not end in a space, assume
-		// span is part of the last word in current split which is not a valid split.
-		if len(split) > 0 && !strings.HasSuffix(split, " ") {
-			shouldSkipAdding = true
+	i := 0
+	parts := []string{""}
+
+	for i < len(query) {
+		// a valid span match is one that starts at the beginning of the query or
+		// after a word separator
+		validSpanStart := true
+		if i > 0 {
+			_, validSpanStart = wordSeparators[query[i-1]]
 		}
```

Comment:
Can we make this a single statement:

```
if i < len(query) && wordSeparators[query[i-1]] {
}
```

If you stored "true" value in wordSeparators, per https://stackoverflow.com/a/2051011 ?

Reply:
done